repository/repo.go

```diff
@@ -155,6 +155,9 @@ type Repo interface {
 	// This is the batch version of the corresponding GetNotes(...) method.
 	GetAllNotes(notesRef string) (map[string][]Note, error)
 
+	// EditNote edit a note to a revision under the given ref.
+	EditNote(ref, revision string, note Note) error
```

Comment:
The git-appraise formats are all based on the idea of an append-only log of events (rather than an editable view of the current state of the world).

That means we should only ever append new notes, not edit existing ones.